lib/rules-generator.js

```diff
@@ -177,9 +176,17 @@ methods(Generator, {
   // Update rules based on the given path expression.
   updateRules: function(path) {
     var i;
-    var pathMethods = ['read', 'write', 'validate'];
+    var pathMethods = ['read', 'write'];
     var location = this.ensurePath(path.parts);
 
+    // Write validation (if non-trivial)
+    location['.validate'] = this.getExpressionText(
+      this.partialEval(this.symbols.functions['@validator@' + path.isType].body));
+    if (location['.validate'] == 'true') {
```

Comment:
I don't want to convert strings to booleans, if that's what you're suggesting.

BTW - this code will likely change to do simplification base on the ast tree rather than the textual representation.  I'd like to provide a simplify() function that will convert:

- validate: true => remove
- write: false => remove
- read: false => remove

as well as simple expression simplifications:

true || exp => true
false || exp => exp
true && exp => exp
false && exp => false

The latter can arise from trivial conditional rules, or when trivial base rules are combined with more complex validation expression in derived Types.